connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -91,7 +92,25 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
         } else {
           child
         }
-      case _ => throw InvalidPlanInput()
+      case proto.Read.ReadTypeCase.DATA_SOURCE =>
+        if (rel.getDataSource.getFormat == "") {
+          throw InvalidPlanInput("DataSource requires a format")
+        }
+        val localMap = CaseInsensitiveMap[String](rel.getDataSource.getOptionsMap.asScala.toMap)
+        if (!localMap.contains("path")) {
+          throw InvalidPlanInput("DataSource requires a path")
+        }
+
+        val reader = session.read
+        reader.format(rel.getDataSource.getFormat)
+        for ((k, v) <- localMap) {
+          reader.option(k, v)
+        }
+        if (rel.getDataSource.getSchema != null) {
+          reader.schema(rel.getDataSource.getSchema)
+        }
+        reader.load().queryExecution.analyzed
```

Comment:
So my question here is, if the execution has already happened or not. What I mean is, is `load()` a blocking operation or a logical one? If you call `load()` and then return the analyzed plan. What happens if you call collect on this plan? Does the load happen again?

Can verify this?

Reply:
`load()` just builds the logical plan. it's not an action.

Reply:
Thanks I was able to verify this for myself as well using some experiments.